commit 721abc529acf2c3b4d8ec7e1420e1b9b68d7ad3b
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Tue Dec 4 19:40:12 2018 +0000

    Revert "staging: vchiq_arm: Register a platform device for the audio driver"
    
    This reverts commit ab59590ed562b89db51fe46cee5db96b9bc5abd8.
    
    Issues have been observed in LibreElec as this was unconditionally
    loading the audio driver instead of having the DT parameter to
    enable it.
    
    Includes a partial revert of 2147700eb7a1b9e55e0684f0749114ce35d61571
    which fixed up the error handling.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c b/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
index 9b751a57c652..40e8e041b781 100644
--- a/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
+++ b/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
@@ -170,7 +170,6 @@ static struct class  *vchiq_class;
 static struct device *vchiq_dev;
 static DEFINE_SPINLOCK(msg_queue_spinlock);
 static struct platform_device *bcm2835_camera;
-static struct platform_device *bcm2835_audio;
 static struct platform_device *bcm2835_codec;
 static struct platform_device *vcsm_cma;
 
@@ -3662,9 +3661,6 @@ static int vchiq_probe(struct platform_device *pdev)
 	bcm2835_camera = vchiq_register_child(pdev, "bcm2835-camera");
 	if (IS_ERR(bcm2835_camera))
 		bcm2835_camera = NULL;
-	bcm2835_audio = vchiq_register_child(pdev, "bcm2835_audio");
-	if (IS_ERR(bcm2835_audio))
-		bcm2835_audio = NULL;
 	bcm2835_codec = vchiq_register_child(pdev, "bcm2835-codec");
 	if (IS_ERR(bcm2835_codec))
 		bcm2835_codec = NULL;
@@ -3685,7 +3681,6 @@ static int vchiq_probe(struct platform_device *pdev)
 static int vchiq_remove(struct platform_device *pdev)
 {
 	platform_device_unregister(bcm2835_codec);
-	platform_device_unregister(bcm2835_audio);
 	platform_device_unregister(bcm2835_camera);
 	platform_device_unregister(vcsm_cma);
 	vchiq_debugfs_deinit();
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1759_721abc529acf2c3b4d8ec7e1420e1b9b68d7ad3b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1759_721abc529acf2c3b4d8ec7e1420e1b9b68d7ad3b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
